src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -28,24 +31,69 @@ def fetch_alert(self, **notification):
 class PrometheusRESTController(RESTController):
     def prometheus_proxy(self, method, path, params=None, payload=None):
         # type (str, str, dict, dict)
+        user = None
+        password = None
+        verify = Settings.PROMETHEUS_API_SSL_VERIFY
+        secure_monitoring_stack = bool(mgr.get_module_option_ex('cephadm', 'secure_monitoring_stack', 'false'))
+        if secure_monitoring_stack:
+            cmd = {'prefix': 'orch prometheus access info'}
+            ret, out, err = mgr.mon_command(cmd)
+            if ret == 0 and out is not None:
+                access_info = json.loads(out)
+                user = access_info['user']
+                password = access_info['password']
+                certificate = access_info['certificate']
+                cert_file = tempfile.NamedTemporaryFile()
+                cert_file.write(certificate.encode('utf-8'))
+                cert_file.flush()
+                verify = cert_file.name
         return self._proxy(self._get_api_url(Settings.PROMETHEUS_API_HOST),
                            method, path, 'Prometheus', params, payload,
-                           verify=Settings.PROMETHEUS_API_SSL_VERIFY)
+                           user=user, password=password, verify=verify)
+
+    def get_config_option(self, key):
+        cmd = {'prefix': 'config get',
+               'who': 'mgr',
+               'key': key}
+        ret, option, err = mgr.mon_command(cmd)
+        if ret < 0 and not ret == -errno.ENOENT:
+            return option
+        return None
 
     def alert_proxy(self, method, path, params=None, payload=None):
         # type (str, str, dict, dict)
+        user = None
+        password = None
+        verify = Settings.ALERTMANAGER_API_SSL_VERIFY
+        secure_monitoring_stack = bool(mgr.get_module_option_ex('cephadm', 'secure_monitoring_stack', 'false'))
+        if secure_monitoring_stack:
+            cmd = {'prefix': 'orch alertmanager access info'}
+            ret, out, err = mgr.mon_command(cmd)
+            if ret == 0 and out is not None:
+                access_info = json.loads(out)
+                user = access_info['user']
+                password = access_info['password']
+                certificate = access_info['certificate']
+                cert_file = tempfile.NamedTemporaryFile()
+                cert_file.write(certificate.encode('utf-8'))
+                cert_file.flush()
+                verify = cert_file.name
```

Comment:
i see a similar code above as well. since both are similar or close to similar, then can we put it inside a function? from looking at it, only the `access info` command differs and we can switch them with variables.

Reply:
> i see a similar code above as well. since both are similar or close to similar, then can we put it inside a function? from looking at it, only the `access info` command differs and we can switch them with variables.

@nizamial09 good idea, I moved the code to a dedicated function `get_access_info`.